web/src/components/widgets/Sidebar.tsx

```diff
@@ -0,0 +1,43 @@
+import { reposStore } from '@/store/repoStore'
+import { NavigationMenu } from '@kobalte/core/navigation-menu'
+import { createSignal, For } from 'solid-js'
+
+const Sidebar = () => {
+  const [isOpen, setIsOpen] = createSignal(false)
+
+  const repos = reposStore.repos.map((repo) => repo.fullName)
+
+  return (
+    <>
+      <button
+        class="bg-accent fixed left-4 top-4 z-10 rounded-full p-2 text-white xl:hidden"
+        onClick={() => setIsOpen(!isOpen())}
+      >
+        - - -
+      </button>
+      <NavigationMenu
+        orientation="vertical"
+        class={`${isOpen() ? 'block' : 'hidden'} bg-background fixed top-[64px] h-full w-[250px] border px-2 py-6 xl:absolute xl:block`}
+      >
+        <NavigationMenu.Menu>
+          <NavigationMenu.Trigger as="a" href="/" class="block">
+            Main Menu
+          </NavigationMenu.Trigger>
+        </NavigationMenu.Menu>
+        <NavigationMenu.Menu>
+          <span class="text-muted-foreground mt-4 block">Repos</span>
+          <NavigationMenu.Separator class="my-2" />
+          <For each={repos}>
+            {(repo) => (
+              <NavigationMenu.Trigger as="a" href={`/${repo}`} class="mt-2 block">
```

Comment:
to handle such navigation it requires the following:
- using A component from solid router (https://docs.solidjs.com/solid-router/reference/components/a), so as prop looks like `as={A}`
- adding a Route component to a router in the tree to handle and render the navigation, it's in App.tsx, here is how to handle params in a router: https://docs.solidjs.com/solid-router/reference/primitives/use-params

the router API is almost equal to react